src/test/groovy/pl/tomaszdziurko/jvm_bloggers/blog_posts/AggregatedRssFeedProducerSpec.groovy

```diff
@@ -55,36 +59,76 @@ class AggregatedRssFeedProducerSpec extends Specification {
             }
         and:
             with (feed.entries[0]) {
-                link == URL + EXPECTED_UTM_SUBSTRING
-                title == TITLE
-                author == AUTHOR
+                link == URL_1 + EXPECTED_UTM_SUBSTRING
+                title == TITLE_1
+                author == AUTHOR_1
                 description.value == DESCRIPTION
                 publishedDate == date
-                author == AUTHOR
+                author == AUTHOR_1
+                uri == UID_1
             }
         and:
             with (feed.entries[1]) {
-                link == URL + EXPECTED_UTM_SUBSTRING
-                title == TITLE
-                author == AUTHOR
+                link == URL_2 + EXPECTED_UTM_SUBSTRING
+                title == TITLE_2
+                author == AUTHOR_2
                 description == null
                 publishedDate == date
-                author == AUTHOR
+                author == AUTHOR_2
+                uri == UID_2
             }
 
     }
 
-    def mockBlogPost(String uid, String description, String title, String url, String author, LocalDateTime date) {
-        BlogPost blogPost = Mock()
-        1 * blogPost.getDescription() >> description
-        1 * blogPost.getTitle() >> title
-        1 * blogPost.getUrl() >> url
-        1 * blogPost.getUid() >> uid 
-        Blog blog = Mock()
-        blog.getAuthor() >> author
-        1 * blogPost.getBlog() >> blog
-        1 * blogPost.getPublishedDate() >> date
-        return blogPost
+    def "Should produce aggregated RSS feed with limited entries count"() {
+        when:
+            SyndFeed feed = rssProducer.getRss(REQUEST_URL, 1)
+        then:
+            Date date = toDate(DATE)
+            with (feed) {
+                links[0].rel == "self"
+                links[0].href == REQUEST_URL
+                feedType == AggregatedRssFeedProducer.FEED_TYPE
+                uri == AggregatedRssFeedProducer.FEED_TITLE
+                title == AggregatedRssFeedProducer.FEED_TITLE
+                description == AggregatedRssFeedProducer.FEED_DESCRIPTION
+                publishedDate == date
+                entries.size() == 1
+            }
+        and:
+            with (feed.entries[0]) {
+                link == URL_1 + EXPECTED_UTM_SUBSTRING
+                title == TITLE_1
+                author == AUTHOR_1
+                description.value == DESCRIPTION
+                publishedDate == date
+                author == AUTHOR_1
+            }
+
+    }
+
+    def "Should throw IAE on blank feedUrl parameter"() {
```

Comment:
Is it really necessary here? Unrolling the test cases will spoil the method name, it will look strange IMHO because input parameters are null, empty and blank string. I think having just a "_blank_" keyword is much more readable method name. But maybe you have another reason behind that?